packages/dds/shared-object-base/src/serializer.ts

```diff
@@ -58,7 +59,103 @@ export interface IFluidSerializer {
 }
 
 /**
- * Data Store serializer implementation
+ * Use for readonly view of Json compatible data that can also contain {@link @fluidframework/core-interfaces#IFluidHandle}s.
+ *
+ * To be "Json compatible" requires that the data is acyclic.
+ *
+ * Note that this does not robustly forbid non json comparable data via type checking,
+ * but instead mostly restricts access to the json compatible subset.
+ *
+ * See also {@link isFluidHandle}.
+ *
+ * Warning: Objects with a key "type" and value "__fluid_handle__" will not round trip correctly through `FluidSerializer`
+ * (due to decodeValue's use of isSerializedHandle causing them to be wrongly parsed as handles) and should be avoided.
+ * @public
+ */
+export type FluidSerializableReadOnly =
```

Comment:
@jason-ha has a PR out about some similar recursive types, is this new type harmonious with whatever he's doing?  I haven't looked as his PR yet.

Reply:
This will not work with the current type test support.
`TypeOnly` utility will not be able to handle anything this such a type in it. (Although currently if this only appears two levels of properties down the error may not be observed.)
See #18523 that injects an interface where there is recursion to workaround the `TypeOnly` issue.